### ISSUE TITLE
Would be good to have nested examples for queries

### ISSUE THREAD

**Issue by vfonic** [samesystem/graphql_rails#77]
I just discovered this gem through rubygems. I really like the simplicity and familiarity of using this for rails developers. :)
Good job! 👍 

I'm interested to see how I can use this to get nested queries like this:

```gql
{
  users(first:10) {
    email
    groups(first:10) {
      title
      description
      members(first:5) {
        email
      }
    }
  }
}
```

Is this automatically available based on ID type (for example from `User has_many :groups, through: :group_memberships`)?

Thanks!

**Comment by povilasjurcys** [comment]
At the moment there is no easy way how to do it. It is only possible to paginate from controller level. That's why there is no example for it. But I see that it would be a very good feature. If nobody opens PR soon, then this will be my top priority feature

**Comment by povilasjurcys** [comment]
hey. I made some investigation. This feature might require to redesign a lot of other gem parts. I'm moving forward, but this might take time. GraphqlRails supports grahhql-ruby types too, so at the moment I would recommend to use plain graphql-ruby types when you need nested pagination:

```ruby
  class GroupType < GraphQL::Schema::Object
    field :shop_id, String, null: false
  end

  class UserType < GraphQL::Schema::Object
    field :groups, GroupType.connection_type, null: false
  end

  class UsersController < GraphqlApplicationController
    action(:index)
      .returns(UserType.connection_type)

    def index
      Users.all
    end
  end
```

**Comment by vfonic** [comment]
What? 😆

Thank you for telling me about `connection_type`! I had no idea this exists. Looks like I missed it from the graphql-ruby documentation.

I'll see what I can do with it. Thanks!

**Comment by povilasjurcys** [comment]
solved with #103
